engine/src/main/java/org/terasology/world/propagation/light/InternalLightProcessor.java

```diff
@@ -64,12 +64,14 @@ private static void populateSunlight(LitChunk chunk) {
         PropagationRules sunlightRules = new SunlightPropagationRules(chunk);
         BatchPropagator lightPropagator = new StandardBatchPropagator(sunlightRules, new SingleChunkView(sunlightRules, chunk));
 
+        Vector3i pos = new Vector3i();
         for (int x = 0; x < ChunkConstants.SIZE_X; x++) {
             for (int z = 0; z < ChunkConstants.SIZE_Z; z++) {
                 for (int y = 0; y < ChunkConstants.MAX_SUNLIGHT; ++y) {
-                    Vector3i pos = new Vector3i(x, y, z);
-                    Block block = chunk.getBlock(x, y, z);
-                    byte light = sunlightRules.getFixedValue(block, pos);
+
+                    byte light = sunlightRules.getFixedValue(
```

Comment:
I'd suggest all memory allocation should occur outside the loops. So, let's get that light variable out of there too. =)